SignalServiceKit/src/Messages/PreKeyBundle+jsonDict.m

```diff
@@ -5,17 +5,27 @@
 #import "NSData+Base64.h"
 #import "PreKeyBundle+jsonDict.h"
 
+NS_ASSUME_NONNULL_BEGIN
+
 @implementation PreKeyBundle (jsonDict)
 
-+ (PreKeyBundle *)preKeyBundleFromDictionary:(NSDictionary *)dictionary forDeviceNumber:(NSNumber *)number {
++ (nullable PreKeyBundle *)preKeyBundleFromDictionary:(NSDictionary *)dictionary forDeviceNumber:(NSNumber *)number
+{
     PreKeyBundle *bundle        = nil;
-    NSString *identityKeyString = [dictionary objectForKey:@"identityKey"];
-    NSArray *devicesArray       = [dictionary objectForKey:@"devices"];
 
-    if (!(identityKeyString && [devicesArray isKindOfClass:[NSArray class]])) {
-        DDLogError(@"Failed to get identity key or messages array from server request");
+    id identityKeyObject = [dictionary objectForKey:@"identityKey"];
+    if (![identityKeyObject isKindOfClass:[NSString class]]) {
```

Comment:
In my testing Android and Desktop didn't crash in this scenario, so presumably they already have more robust parsing logic than we did.